### PR TITLE
gh-125714: Doc: Explain that multiprocessing only supports unpicklable functions through the fork method

### PR DESCRIPTION
**Pull request by progval**
In multiprocessing and concurrent.futures's documentations, as well as the release notes for Python 3.14.

This moves the paragraph in 3.14's release notes from the deprecations section to 'Porting to Python 3.14' because it is a breaking change, which follows a deprecation in Python 3.12.

This explicitly mentions `PicklingError` in this section, so it is easy to find for somehow who gets the error and searches its name through the page.

* Issue: gh-125714

----
📚 Documentation preview 📚: https://cpython-previews--125750.org.readthedocs.build/

### CODE REVIEW

**Comment by progval** [comment]
Is it possible to skip the CLA for such a small change?

**Comment by rruuaanng** [comment]
You must sign the CLA, which gives us permission to distribute your work.

File: Doc/whatsnew/3.14.rst [hunk 1]
```diff
@@ -643,6 +637,17 @@ Changes in the Python API
   This temporary change affects other threads.
   (Contributed by Serhiy Storchaka in :gh:`69998`.)
 
+* :mod:`multiprocessing` and :mod:`concurrent.futures`:
+  The default start method (see :ref:`multiprocessing-start-methods`) changed
+  away from *fork* to *forkserver* on platforms where it was not already
+  *spawn* (Windows & macOS). If you require the threading incompatible *fork*,
```
Comment (JelleZijlstra) [inline Doc/whatsnew/3.14.rst]: Might be better to mention the platforms where it *did* change.

**Review by gpshead — APPROVED** [review]
Thanks!

File: Doc/whatsnew/3.14.rst [hunk 2]
```diff
@@ -643,6 +637,19 @@ Changes in the Python API
   This temporary change affects other threads.
   (Contributed by Serhiy Storchaka in :gh:`69998`.)
 
+* :mod:`multiprocessing` and :mod:`concurrent.futures`:
+  The default start method (see :ref:`multiprocessing-start-methods`) changed
+  away from *fork* to *forkserver* on POSIX platforms which support passing file
+  descriptors over Unix pipes such as Linux. Windows and macOS are keep *spawn*
```
Comment (JelleZijlstra) [inline Doc/whatsnew/3.14.rst]:
```suggestion
  descriptors over Unix pipes, such as Linux. Windows and macOS keep *spawn*
```

File: Doc/whatsnew/3.14.rst [hunk 3]
```diff
@@ -643,6 +637,19 @@ Changes in the Python API
...
+  as their default method.
+  If you require the threading incompatible *fork* start method,
+  which can run :ref:`unpicklable functions <pickle-picklable>`,
+  you must explicitly specify it when using :mod:`multiprocessing`
+  or :mod:`concurrent.futures` APIs.
+  Code running pickable functions on these platforms without explicitly opting
```
Comment (JelleZijlstra) [inline Doc/whatsnew/3.14.rst]:
```suggestion
  Code running picklable functions on these platforms without explicitly opting
```

File: Doc/whatsnew/3.14.rst [hunk 4]
```diff
@@ -643,6 +637,19 @@ Changes in the Python API
   This temporary change affects other threads.
   (Contributed by Serhiy Storchaka in :gh:`69998`.)
 
+* :mod:`multiprocessing` and :mod:`concurrent.futures`:
+  The default start method (see :ref:`multiprocessing-start-methods`) changed
+  away from *fork* to *forkserver* on POSIX platforms which support passing file
+  descriptors over Unix pipes such as Linux. Windows and macOS are keep *spawn*
+  as their default method.
+  If you require the threading incompatible *fork* start method,
```
Comment (JelleZijlstra) [inline Doc/whatsnew/3.14.rst]:
```suggestion
  If you require the threading-incompatible *fork* start method,
```

**Comment by JelleZijlstra** [comment]
I am afraid you will have to sign the CLA.

**Comment by progval** [comment]
> You must sign the CLA, which gives us permission to distribute your work.

I already did by submitting this PR after accepting [Github's terms of service](https://docs.github.com/en/site-policy/github-terms/github-terms-of-service#6-contributions-under-repository-license):

> Whenever you add Content to a repository containing notice of a license, you license that Content under the same terms, and you agree that you have the right to license that Content under those terms.

Nonetheless, I(m fine with the CLA on principle, but I'm not willing to give out the personal info the cla-bot or the form are asking for.

**Comment by willingc** [comment]
I'm going to go ahead and close this PR if there is no signed CLA. If someone wishes to work with @progval to take over this work, we can reopen the PR. Thanks @progval for your effort.

**Pull request closed without merging**